app/assets/javascripts/darkswarm/services/map.js.coffee

```diff
@@ -1,21 +1,42 @@
-Darkswarm.factory "OfnMap", (Enterprises, EnterpriseModal) ->
+Darkswarm.factory "OfnMap", (Enterprises, EnterpriseModal, MapConfiguration) ->
   new class OfnMap
     constructor: ->
-      @enterprises = @enterprise_markers(Enterprises.enterprises)
+      @coordinates = {}
+      @enterprises = Enterprises.enterprises.filter (enterprise) ->
+        # Remove enterprises w/o lat or long
+        enterprise.latitude != null || enterprise.longitude != null
+      @enterprises = @enterprise_markers(@enterprises)
+      self = this
       @enterprises = @enterprises.filter (enterprise) ->
-        enterprise.latitude != null || enterprise.longitude != null # Remove enterprises w/o lat or long
+        # Remove enterprises w/o lat or long
+        enterprise.latitude != null || enterprise.longitude != null
```

Comment:
this filter is now duplicating the one done at line 5, right?

Reply:
Will remove this :)